.buildkite/hooks/pre-command

```diff
@@ -12,10 +33,17 @@ if [[ "$BUILDKITE_PIPELINE_SLUG" == "filebeat" || "$BUILDKITE_PIPELINE_SLUG" ==
   fi
 fi
 
-if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats-metricbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-libbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-packetbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-winlogbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-libbeat" ]]; then
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats-metricbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-libbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-packetbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-winlogbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-libbeat" || "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-metricbeat" ]]; then
   source .buildkite/scripts/setenv.sh
   if [[ "${BUILDKITE_COMMAND}" =~ ^buildkite-agent ]]; then
     echo "Skipped pre-command when running the Upload pipeline"
     exit 0
   fi
 fi
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "beats-xpack-metricbeat" && "$BUILDKITE_STEP_KEY" == "extended-cloud-test" ]]; then
+  BEATS_AWS_SECRET_KEY=$(retry 5 vault kv get -field secret_key ${AWS_SERVICE_ACCOUNT_SECRET_PATH})
+  export BEATS_AWS_SECRET_KEY
+  BEATS_AWS_ACCESS_KEY=$(retry 5 vault kv get -field access_key ${AWS_SERVICE_ACCOUNT_SECRET_PATH})
+  export BEATS_AWS_ACCESS_KEY
+fi
```

Comment:
After thinking a bit more about `defineModuleFromTheChangeSet` and how other places could use it. I think `MODULE` could be set in the `pre-command` hook. 


```suggestion

# Set the MODULE env variable
defineModuleFromTheChangeSet "${BEATS_PROJECT_NAME}"
```

Therefore my former suggestion for adding `defineModuleFromTheChangeSet` in `.buildkite/scripts/cloud_tests.sh` could be deleted.

Reply:
I tested it here: https://github.com/elastic/beats/pull/38081/commits/c96b847d7170701b89840b4b497985b438166273
And if we move that right now to the `pre-command`, it will require moving to `pre-command` all dependencies, answered [here](https://github.com/elastic/beats/pull/38081#discussion_r1519462128)

Reply:
Then, with this approach, the `MODULE` env variable won't be set in the other stages. Hence, it will take longer to run the tests.

If that's something to be done in phase2 that's totally fine with me. But PRs will be slow to be tested. Unless the `defineModuleFromTheChangeSet "${BEATS_PROJECT_NAME}"` is also called in all those tests using the `withModule: true`